business_objects/monitor.py

```diff
@@ -0,0 +1,285 @@
+from typing import Any, List, Optional
+from . import general
+from .. import enums
+
+
+def get_all_tasks(
+    project_id: Optional[str] = None,
+    only_running: bool = True,
+    limit_per_task: int = 100,
+) -> List[Any]:
+    query = f""" 
+    SELECT tasks.*, p.name project_name, orga.name organization_name
+    FROM (
+        ({__select_running_information_source_payloads(project_id, only_running, limit_per_task)})
+        UNION ALL
+        ({__select_running_attribute_calculation_tasks(project_id, only_running, limit_per_task)})
+        UNION ALL
+        ({__select_running_tokenization_tasks(project_id, only_running, limit_per_task)})
+        UNION ALL
+        ({__select_running_embedding_tasks(project_id, only_running, limit_per_task)})
+        UNION ALL
+        ({__select_running_weak_supervision_tasks(project_id, only_running, limit_per_task)})
+        UNION ALL
+        ({__select_running_upload_tasks(project_id, only_running, limit_per_task)})
+    ) tasks
+    INNER JOIN project p
+        ON p.id = tasks.project_id
+    INNER JOIN organization orga
+        ON orga.id = p.organization_id
+    """
+
+    return general.execute_all(query)
+
+
+def cancel_all_running_tasks(project_id: str = None):
+    set_information_source_payloads_to_failed(project_id)
+    set_attribute_calculation_to_failed(project_id)
+    set_embedding_to_failed(project_id)
+    set_record_tokenization_task_to_failed(project_id)
+    set_upload_task_to_failed(project_id)
+    set_weak_supervision_to_failed(project_id)
+    general.commit()
+
+
+def set_information_source_payloads_to_failed(
+    project_id: Optional[str] = None,
+    payload_id: Optional[str] = None,
+    with_commit: bool = False,
+) -> None:
+    query = f"""
+    UPDATE {enums.Tablenames.INFORMATION_SOURCE_PAYLOAD.value}
+    SET state = '{enums.PayloadState.FAILED.value}'
+    WHERE state = '{enums.PayloadState.CREATED.value}'
+    """
+
+    if project_id:
+        query = query + f""" AND project_id = '{project_id}'"""
+
+    if payload_id:
+        query = query + f"""AND id = '{payload_id}'"""
```

Comment:
i dont think this works since no space is provided so project_id and payload_id set would result in an issue.
Also, """ shouldn't be necessary for single lines.

Reply:
We could also use the helper method so we don't need to worry about spaces
```suggestion
    if project_id:
        query = __extend_where_helper(query, f"project_id = '{project_id}'"

    if payload_id:
        query = __extend_where_helper(query,f"id = '{payload_id}'"
```